### PR TITLE
OpenAI Feed: use separator > for nested product_category

### PR DESCRIPTION
**Pull request by htdat**
Fixes: WOOPTP-47

## Description

Improves the `ProductMapper` to correctly handle product categories for variations by using parent product data.

## Changes:

1. **Product category path improvements**: The `get_product_category()` method now uses parent product categories for variations and ensures correct hierarchical ordering with " > " separator. The method finds the deepest category, builds the path in correct order, and handles edge cases.

2. **Add tests for enable_search/enable_checkout with parent product**: Added new unit tests to verify that variations correctly use their parent product's meta settings for `enable_search` and `enable_checkout` fields.

## Testing instructions

Create a simple or variable product with nested categories, and confirm: 
- If nested category, the deepest one is selected.
- The category path uses the correct separator `>`. 

## Checklist

- [x] `npm run test:php` does not return errors.
- [x] `npm run lint:php` does not return errors.

## Summary by CodeRabbit

* **Refactor**
  * Improved category hierarchy logic to determine the deepest category and build ordered hierarchical category paths for products and variations.
* **Bug Fixes**
  * Variations inherit search and checkout settings from their parent; handles cases with multiple/no categories correctly.
* **Tests**
  * Added extensive tests for category path construction, deepest-category selection, variation inheritance, single-level/no-category cases, and cleanup.

### CODE REVIEW

File: src/Integrations/OpenAi/ProductMapper.php [hunk 1]
```diff
@@ -330,21 +330,44 @@ private function generate_mpn( \WC_Product $product ): string {
...
+				'fields'   => 'id=>name',
+				'taxonomy' => 'product_cat',
+			]
+		);
 
-		return empty( $names ) ? null : implode( ', ', $names );
+		return empty( $category_names )
+			? null
+			: implode( ' > ', $category_names );
 	}
```
[the remark this answers is not in the document]
Comment (htdat) [reply src/Integrations/OpenAi/ProductMapper.php]: should be fixed in 3cd4d509f5747005f90910f172a3a241802fe4a1

File: src/Integrations/OpenAi/ProductMapper.php [hunk 2]
```diff
@@ -218,7 +218,7 @@ private function get_enable_with_override( \WC_Product $product, string $meta_ke
 	 */
 	protected function get_enable_search( \WC_Product $product, ?\WC_Product $parent_product ): string {
 		// For variations, check parent product meta; for simple products, check product meta.
-		$check_product = $parent_product ? $parent_product : $product;
+		$check_product = $parent_product ?? $product;
```
Comment (RadoslavGeorgiev) [inline src/Integrations/OpenAi/ProductMapper.php:221]: Good point. Before stronger types, we were at risk of `false ?? $product` resulting in `false`, but we have strong types now!

File: src/Integrations/OpenAi/ProductMapper.php [hunk 3]
```diff
@@ -330,21 +330,74 @@ private function generate_mpn( \WC_Product $product ): string {
...
+		 *
+		 * @param array<int, string> $category_names Arrays with key is category_id, value is category name.
+		 */
+		$category_names = get_terms(
+			[
+				'include'  => $ordered_ids,
+				'fields'   => 'id=>name',
+				'taxonomy' => 'product_cat',
+			]
+		);
```
Comment (RadoslavGeorgiev) [inline src/Integrations/OpenAi/ProductMapper.php]: Considering those terms have already been cached in memory, this remains the only piece that performs additional queries. If you replace it with the suggestion, there are no addition DB queries for categories:


```suggestion
        $category_names = [];
		foreach ( $ordered_ids as $term_id ) {
			$term = get_term( $term_id, 'product_cat' );
			if ( $term && ! is_wp_error( $term ) ) {
				$category_names[ $term_id ] = $term->name;
			}
		}
```

**Review by RadoslavGeorgiev — COMMENTED** [review]
Reviewing this PR made me feel something I hadn't felt in a while, like I'm on a roller coaster! 🤣  

At first I thought that each `$product->get_category_ids()`, `get_ancestors()` and `get_terms()` call would be a performance catastrophe. However, it turns out that:

1. `wc_get_products()` caches the products' categories.
2. `get_ancestors()` uses `get_term()` internally, and that hits the cache, populated by `wc_get_products()`.
3. `get_terms()` is actually the only addition in this PR that reaches out to the database. However, as you will see in one of my comments, that can easily be replaced with `get_term()`.

When you draw the line, we can generate this list without a single additional DB query! 😳

Once you make the change, I'm curious if you can xDebug it and confirm my findings. I used a breakpoint in the beginning of the `get_product_category()` and simply visited the edit screen of a product with nested categories.

File: src/Integrations/OpenAi/ProductMapper.php (hunk 3 above)
Comment (htdat) [reply src/Integrations/OpenAi/ProductMapper.php]: Fixed it here fb78b920d5640f10562a7f6b466458e6e7a63a6d :D

**Comment by htdat** [comment]
@RadoslavGeorgiev :D 

> get_terms() is actually the only addition in this PR that reaches out to the database. However, as you will see in one of my comments, that can easily be replaced with get_term().

You're absolutely right. I tried to optimize this, but then it turns out all of them have been cached already, so calling `get_term` is more efficient in the end at `WP_Term::get_instance` with: `$_term = wp_cache_get( $term_id, 'terms' );`

**Review by RadoslavGeorgiev — APPROVED** [review]
Thanks for improving the performance! :shipit:
